CRM/Upgrade/Incremental/sql/5.56.alpha1.mysql.tpl

```diff
@@ -1,5 +1,23 @@
 {* file to handle db changes in 5.56.alpha1 during upgrade *}
 
+-- Add in Year prior to previous fiscal year and Previous 2 fiscal years
+SELECT @option_group_id_date_filter := max(id) from civicrm_option_group where name = 'relative_date_filters';
+UPDATE `civicrm_option_value` SET weight = 64 WHERE option_group_id = @option_group_id_date_filter AND weight = 62;
```

Comment:
I was wrong about the insert ignore because this table allows two with the same name, so if someone reruns the upgrade you get two (or more) copies of the rows. Also these update statements are going to keep shifting the weight each time the upgrade is run. Or if someone has manually rearranged then the insert might not be in the "right" place.

So after all that wrangling:
(a) While nice-to-have, given that someone can just add their own option values and it's documented, how much more time do you want to spend fighting with it?
(b) You could do this in php using [ensureOptionValueExists](https://github.com/civicrm/civicrm-core/blob/a715421290d4afc8b6998e342d3378a35212597c/CRM/Upgrade/Incremental/php/FiveFiftyThree.php#L73), and [this multilingual technique](https://github.com/civicrm/civicrm-core/blob/8c33f6c17c15b6cd7ae30f3f9a5959834700c7c7/CRM/Upgrade/Incremental/php/FiveTwentyNine.php#L73-L84) and maybe just leave the weight updating out and insert at max(weight)+1.

Reply:
Yes, I see this could get messy. I did wonder a bit about the order shifting. Thanks for helping out with this even though it's pretty unimportant (but we've come this far...)

ensureOptionValueExists does seem like a much better way to go.

For the order, I've done some experimenting and if you specify a weight in API4, it cleverly increments all the higher weight options, so nothing else needs to be changed. I can update ensureOptionValueExists to use API4. I could also add a param called insert_after, which looks up the option we want to insert after and then sets the weight to one higher (or does not set a weight if we don't find the option, in which case it gets added as max weight + 1.

That leaves the multilingual part. How about adding that multilingual logic to ensureOptionValueExists, so this is easier in the future?
If we just created an option value and we're multilingual, for each locale, update the row with the translated label and description.

Reply:
Or probably better just to create a new function that uses API4 and does localization and leave this one as is to ensure compatibility.

Reply:
If you want to split this into 2 PRs - one that only adds it on new installs and then a separate for the upgrade, I think it would be an easy merge for the first one.

Also FYI there's some open questions about how much api should be used in upgrade steps - see [here](https://docs.civicrm.org/dev/en/latest/framework/upgrade/#tips-prefer-simple-sql-semantics-over-apibaodao) and [here](https://github.com/civicrm/civicrm-core/pull/24364).

Reply:
OK, keep it low-level. So it turns out there's actually a [readme](https://github.com/civicrm/civicrm-core/blob/master/CRM/Upgrade/Incremental/sql/README.txt) that specifically covers how to add option values (I've added a link to this in the docs).

But the SQL in there still doesn't deal with the case where the option already exists, so I've done that in SQL as well. If that makes sense, I can update the readme with this as an example.

I can still split this — or maybe it's getting there now?